qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -893,6 +895,15 @@ def _run_analysis(
                         "ydata": series_data.y,
                         "sigma": series_data.y_err,
                     }
-                analysis_result["raw_data"] = raw_data_dict
+                result_data["raw_data"] = raw_data_dict
+
+        analysis_result = AnalysisResultV1(
+            result_data=result_data,
+            result_type=result_data["analysis_type"],
+            device_components=[
+                Qubit(qubit) for qubit in experiment_data.data(0)["metadata"]["qubits"]
```

Comment:
Is there a new way to retrieve the qubit, through experiment metadata?

Reply:
I don't think qubit is currently added to experiment metadata, but it'd be a good thing to do. It'd certainly make this line easier.